### ISSUE TITLE
Deprecate `CreateChildSpan` to start spans from in `github.com/coopnorge/go-datadog-lib/tracing`

### ISSUE THREAD

**Issue by nhhagen** [coopnorge/go-datadog-lib#409]
`CreateChildSpan` will create a child-span of the span embedded in sourceCtx. If there is no trace-information in sourceCtx, a noop-span will be returned. 

https://github.com/coopnorge/go-datadog-lib/blob/83efedb779f2719d027c6eac5c71465800f5dd61/tracing/tracing.go#L27-L39

The functionality is nearly identical with `StartSpanFromContext` from `gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer` and I belive it also serves the same purpose. Datadogs `StartSpanFromContext` also returs a `context.Context` so that further child spans can be created.

https://github.com/DataDog/dd-trace-go/blob/ac73f9bac78109a603d938c3a67bfa55e2969062/ddtrace/tracer/context.go#L36-L60

```go
// StartSpanFromContext returns a new span with the given operation name and options. If a span
// is found in the context, it will be used as the parent of the resulting span. If the ChildOf
// option is passed, it will only be used as the parent if there is no span found in `ctx`.
func StartSpanFromContext(ctx context.Context, operationName string, opts ...StartSpanOption) (Span, context.Context) {
	// copy opts in case the caller reuses the slice in parallel
	// we will add at least 1, at most 2 items
	optsLocal := make([]StartSpanOption, len(opts), len(opts)+2)
	copy(optsLocal, opts)

	if ctx == nil {
		// default to context.Background() to avoid panics on Go >= 1.15
		ctx = context.Background()
	} else if s, ok := SpanFromContext(ctx); ok {
		optsLocal = append(optsLocal, ChildOf(s.Context()))
	}
	optsLocal = append(optsLocal, withContext(ctx))
	s := StartSpan(operationName, optsLocal...)
	if span, ok := s.(*span); ok && span.pprofCtxActive != nil {
		// If pprof labels were applied for this span, use the derived ctx that
		// includes them. Otherwise a child of this span wouldn't be able to
		// correctly restore the labels of its parent when it finishes.
		ctx = span.pprofCtxActive
	}
	return s, ContextWithSpan(ctx, s)
}
```

I suggest we deprecate `github.com/coopnorge/go-datadog-lib/tracing.CreateChildSpan` and point users of it to `gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer.StartSpanFromContext`

**Comment by bendiknesbo** [comment]
The differences between `gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer.StartSpanFromContext` and `github.com/coopnorge/go-datadog-lib/tracing.CreateChildSpan` is that:
- `gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer.StartSpanFromContext` will return a new context with the span, so that another call to `StartSpanFromContext` will create a nested span.
- `github.com/coopnorge/go-datadog-lib/tracing.CreateChildSpan` will not create a span unless the `sourceCtx` already contains a span.

So deprecating `CreateChildSpan` might cause a lot more top-level traces to be created. That might be valid for some workloads (like cron-jobs, where you want to trace every run), but invalid for others (like APIs, where there is some sampling-mechanic before reaching the API-handler)

**Comment by bendiknesbo** [comment]
`github.com/coopnorge/go-datadog-lib/tracing.CreateChildSpan` should definitely have returned a context.

**Comment by nhhagen** [comment]
There is almost no use of `CreateChildSpan` https://github.com/search?q=org%3Acoopnorge%20CreateChildSpan&type=code.

What to do here depends on what we want to achieve. Either we deprecate and point to `gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer.StartSpanFromContext` or we treat this as a bug and introduce the correct implementation, deprecate the old and point it to the new one. The documentation should make it very clear what the difference of these functions are.

**Comment by bendiknesbo** [comment]
@arunpoudel and @LinMAD , do you have any thoughts here?
Your repo's are the only one using `CreateChildSpan`, either as a net/http.Server middleware (this can be included into go-datadog-lib, or we could use https://github.com/DataDog/dd-trace-go/blob/main/contrib/net/http/http.go directly), or to start spans for SAP-calls.

I'm also noticing different behaviours in the different middleware-implementations from dd-trace-go:
- `database/sql` can create a new span for all calls, or only if a parent span exists: https://github.com/DataDog/dd-trace-go/blob/main/contrib/database/sql/conn.go#L329-L331
- `net/http.RoundTripper` needs to implement a custom `ignoreRequest` to do the same: https://github.com/DataDog/dd-trace-go/blob/main/contrib/net/http/roundtripper.go#L29-L31
- `google.golang.org/grpc` **always** creates spans: https://github.com/DataDog/dd-trace-go/blob/main/contrib/google.golang.org/grpc/client.go#L168-L200

**Comment by arunpoudel** [comment]
I think we are going to need `net/http.Server` middleware from this library itself. Althought https://github.com/DataDog/dd-trace-go/blob/main/contrib/net/http/http.go#L85 already provides this functionality, it would be nice to have one in our package, so, that I do not have to worry about the injecting service name etc.
